### PR TITLE
disttask: replace manually mock with gomock

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: ref #41495

Problem Summary:
replace manually mock with mock generated by `gomock`.

Note: do **`make gen_mock`** after any of the mocked interface changed
### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: disttask/framework/scheduler/manager_test.go [hunk 1]
```diff
@@ -150,44 +169,66 @@ func TestManager(t *testing.T) {
...
-	mockInternalScheduler.On("Run", mock.Anything, task1).Return(nil).Once()
-	mockTaskTable.On("HasSubtasksInStates", id, taskID1, proto.StepOne, []interface{}{proto.TaskStatePending, proto.TaskStateRevertPending}).Return(false, nil)
-	mockInternalScheduler.On("Stop").Once()
+	mockTaskTable.EXPECT().HasSubtasksInStates(id, taskID1, proto.StepOne,
+		[]interface{}{proto.TaskStatePending, proto.TaskStateRevertPending}).
+		Return(true, nil)
+	var wg sync.WaitGroup
+	var taskCount int
+	runFn := func(f func()) error {
+		taskCount++
```
Comment (ywqzzy) [inline disttask/framework/scheduler/manager_test.go:191]: seems useless？

File: disttask/framework/scheduler/manager_test.go [hunk 2]
```diff
@@ -93,54 +97,69 @@ func TestOnRunnableTasks(t *testing.T) {
...
-	mockTaskTable.AssertExpectations(t)
-	mockInternalScheduler.AssertExpectations(t)
-	mockPool.AssertExpectations(t)
+	m.onRunnableTasks(context.Background(), []*proto.Task{task})
+
+	wg.Wait()
 }
 
 func TestManager(t *testing.T) {
 	// TODO(gmhdbjd): use real subtask table instead of mock
```
Comment (ywqzzy) [inline disttask/framework/scheduler/manager_test.go:157]: remove the todo?

File: disttask/framework/scheduler/manager_test.go [hunk 3]
```diff
@@ -17,13 +17,15 @@ package scheduler
 import (
 	"context"
 	"errors"
+	"sync"
 	"testing"
 	"time"
 
+	"github.com/golang/mock/gomock"
+	"github.com/pingcap/tidb/disttask/framework/mock"
```
Comment (hawkingrei) [inline disttask/framework/scheduler/manager_test.go:25]: we have migrated it to ```go.uber.org/mock/gomock```.

Comment (D3Hunter) [reply disttask/framework/scheduler/manager_test.go:25]: seems no code is using this one, let's do it later

File: disttask/framework/scheduler/manager_test.go [hunk 4]
```diff
@@ -17,16 +17,30 @@ package scheduler
...
 	"github.com/pingcap/tidb/disttask/framework/proto"
 	"github.com/pingcap/tidb/resourcemanager/pool/spool"
 	"github.com/pingcap/tidb/resourcemanager/util"
-	"github.com/stretchr/testify/mock"
 	"github.com/stretchr/testify/require"
+	"go.uber.org/mock/gomock"
 )
 
+func getPoolRunFn() (*sync.WaitGroup, func(f func()) error) {
+	wg := &sync.WaitGroup{}
```
Comment (hawkingrei) [inline disttask/framework/scheduler/manager_test.go:33]: use ```util.WaitGroupWrapper``` ?

Comment (D3Hunter) [reply disttask/framework/scheduler/manager_test.go:33]: no，current is enough

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Review by ywqzzy — APPROVED** [review]
LGTM

**Pull request merged**